engine-tests/src/test/java/org/terasology/rendering/dag/StateManagerTest.java

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag;
+
+
+import java.util.Arrays;
+import java.util.Collection;
+import org.junit.Test;
+import org.terasology.rendering.dag.states.AbstractState;
+import org.terasology.rendering.dag.states.StateType;
+import org.terasology.rendering.dag.states.StateValue;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+public class StateManagerTest {
+    @Test
+    public void testOneStateTwoNodes() {
+        StateManager stateManager = new StateManager();
+        DummyState dummyStateA = new DummyState();
+        stateManager.addState(DummyStateType.A, dummyStateA);
+        DummyNode node1 = instantiateDummyNodeWithState("node1", DummyStateType.A, StateValue.ENABLED);
+        DummyNode node2 = instantiateDummyNodeWithState("node2", DummyStateType.A, StateValue.ENABLED);
```

Comment:
As a general comment on testing, tests are supposed to be part of the documentation on how to use the tested code. Not quite as far as a tutorial, but representative of production code nevertheless. At least that's the theory.

In this case I'd therefore keep render graph instantiation, node instantiation and wiring between the two in very explicit form rather than wrapping them within private methods such as instantiateDummyNodeWithState and createDummyPipeline. Unless you can create a single "setup(void)" method that does all this for many test cases. On that base you can then customize nodes and graph depending on need, i.e.:

```
commonTestSetup()
nodeA.customizeColor(blue)
nodeB.customizeColor(red)
...etc, etc...
```

I'd also consider what is representative in terms of instantiation timeline. For example, the render graph is among the first classes to be instantiated in this context. I'd therefore instantiate that first. I'd then instantiate and add nodes to it. (I remind you those instantiations might be in a setup method. Finally I'd deal with the focus of the testing, that is the StateManager instantiation and its interactions with the other objects. This would lead to all "focus" code, include its testing asserts, all close to each other and without distractions due to setting up other instances. This way you get a macrostructure of a test such as:

- common setup
- setup customization for specific test
- focus code
- actual testing

Having said that, don't rush to change this specific test class as I'm not sure the class it tests will survive scrutiny in its current form. So, consider what I wrote only in general terms, for future test classes.